smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -86,14 +91,17 @@ contract MixinPurchase is
       // This is an existing owner trying to extend their key
       newTimeStamp = toKey.expirationTimestamp + expirationDuration;
       toKey.expirationTimestamp = newTimeStamp;
+
+      // set key manager
+      _setKeyManagerOf(idTo, _keyManager);
```

Comment:
So actually in tat case, we should NOT reset the key manager.

Reply:
yep makes sense as we are not using `msg.sender` here